### PR TITLE
Addition of Cloud SQL Proxy for Postgres

### PR DESCRIPTION
**Pull request by adamstrawson**
Introducing Cloud SQL Proxy as a sidecar container for services who have `infrastructure.postgres.enabled` set. 

- Resources probably need to be tweaked, I've matched them to the "main" container as a starting point, but once we have metrics to understand usage we can adjust. 
- addition in `values.yaml` is a temp workaround, I imagine we'll store these cluster specific values in a configmap once we roll out [config-syncer](https://app.shortcut.com/sequence/story/452/set-up-config-syncer)

Tests;

Pod running with 2 containers (Service & Proxy Sidecar) 
```
adam-proxy-test-b8c557887-zjgkj     2/2     Running   0          33s
```

Logs for Cloud SQL Proxy, shows working connections

```
2022/02/01 17:18:09 Listening on 127.0.0.1:5432 for non-production-339214:europe-west2:non-production
2022/02/01 17:18:09 Ready for new connections
2022/02/01 17:18:11 Generated RSA key in 1.76143308s
2022/02/01 17:18:13 New connection for "non-production-339214:europe-west2:non-production"
2022/02/01 17:18:13 refreshing ephemeral certificate for instance non-production-339214:europe-west2:non-production
2022/02/01 17:18:13 Scheduling refresh of ephemeral certificate in 54m59.725035174s
2022/02/01 17:18:14 Client closed local connection on 127.0.0.1:5432
```

Logs for Service (`counterparty`) connecting to the proxy via `localhost` and running the migration
```
17:18:14.154 [main] INFO  o.f.c.i.license.VersionPrinter - Flyway Community Edition 8.4.2 by Redgate
17:18:14.158 [main] INFO  o.f.c.i.d.base.BaseDatabaseType - Database: jdbc:postgresql://localhost:5432/dev-counterparty (PostgreSQL 13.4)
17:18:14.666 [main] INFO  o.f.core.internal.command.DbValidate - Successfully validated 3 migrations (execution time 00:00.209s)
17:18:14.756 [main] INFO  o.f.core.internal.command.DbMigrate - Current version of schema "public": 3
17:18:14.758 [main] INFO  o.f.core.internal.command.DbMigrate - Schema "public" is up to date. No migration necessary.
17:18:20.952 [DefaultDispatcher-worker-1] INFO  ktor.application - Autoreload is disabled because the development mode is off.
17:18:21.055 [DefaultDispatcher-worker-1] INFO  ktor.application - Responding at http://0.0.0.0:9000
17:18:21.056 [DefaultDispatcher-worker-1] INFO  ktor.application - Application started in 0.601 seconds.
17:18:21.359 [DefaultDispatcher-worker-2] INFO  ktor.application - Autoreload is disabled because the development mode is off.
17:18:21.360 [DefaultDispatcher-worker-2] INFO  ktor.application - Responding at http://0.0.0.0:9001
17:18:21.360 [DefaultDispatcher-worker-2] INFO  ktor.application - Application started in 0.098 seconds.
```

### CODE REVIEW

File: values.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ deployment:
 infrastructure:
   postgres:
     enabled: false
-    instanceIP: "10.136.0.3"
+    connectionName: non-production-339214:europe-west2:non-production
```
Comment (grrywlsn) [inline values.yaml:11]: imo this could be picked up from a `cluster-defaults`-type ConfigMap in future, but this looks good

Comment (grrywlsn) [reply values.yaml:11]: nm just saw your comment, you're on it. 🔥

**Comment by grrywlsn** [comment]
the database credentials (expect the password) were also still passed in from the ConfigMap (which is mounted and copied to the `.env` file): https://github.com/SequenceHQ/helm-api/blob/1.4.2/templates/configmap-dotenv.yaml#L11

Now that we have `DATABASE_URL` and I think that has been accepted as a standard, perhaps we just remove all the Postgres stuff from the ConfigMap.  Worth checking in Slack if we can remove it.

**Comment by adamstrawson** [comment]
> the database credentials (expect the password) were also still passed in from the ConfigMap (which is mounted and copied to the `.env` file): https://github.com/SequenceHQ/helm-api/blob/1.4.2/templates/configmap-dotenv.yaml#L11
> 
> Now that we have `DATABASE_URL` and I think that has been accepted as a standard, perhaps we just remove all the Postgres stuff from the ConfigMap. Worth checking in Slack if we can remove it.

Thanks @grrywlsn, confirmed that the new standard is being used (only Counterparty is using postgres so far), so I've removed the old env vars in favour of just `DATABASE_URL`

**Review by emalihin — APPROVED** [review]

**Review by grrywlsn — APPROVED** [review]
ship it 🛥️

**Pull request merged**